### modelo predictivo para la deteccion de paros cardiacos

### 1. Carga y exploracion inicial del data frame

In [38]:
# Librerias de visualizacion y importacion de datos.abs
import pandas as pd 
import matplotlib.pyplot as plt 
import seaborn as sns 

df = pd.read_csv("../data/raw/HeartAttackDataSet.csv")

# Dimenciones del data frame 
print(f"=" * 80)
print(f"Dimensiones del data frame: {df.shape[0]} numero de filas y {df.shape[1]} numero de columnas ")
# Información del data frame
print(f"=" * 80)
print (f"Tipos de datos del data frame")
print(f"=" *80)
df.info()


Dimensiones del data frame: 303 numero de filas y 14 numero de columnas 
Tipos de datos del data frame
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 303 entries, 0 to 302
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       303 non-null    int64  
 1   sex       303 non-null    int64  
 2   cp        303 non-null    int64  
 3   trestbps  303 non-null    int64  
 4   chol      303 non-null    int64  
 5   fbs       303 non-null    int64  
 6   restecg   303 non-null    int64  
 7   thalach   303 non-null    int64  
 8   exang     303 non-null    int64  
 9   oldpeak   303 non-null    float64
 10  slope     303 non-null    int64  
 11  ca        303 non-null    int64  
 12  thal      303 non-null    int64  
 13  target    303 non-null    int64  
dtypes: float64(1), int64(13)
memory usage: 33.3 KB


In [39]:
# Primeras 5 filas del data frame
print(f"Primeras 5 filas del data frame")
df.head()
# Validacion de tipos de datos del data frame 

Primeras 5 filas del data frame


,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
0,63,1,3,145,233,1,0,150,0,2.3,0,0,1,1
1,37,1,2,130,250,0,1,187,0,3.5,0,0,2,1
2,41,0,1,130,204,0,0,172,0,1.4,2,0,2,1
3,56,1,1,120,236,0,1,178,0,0.8,2,0,2,1
4,57,0,0,120,354,0,1,163,1,0.6,2,0,2,1


### 2. Analisis de calidad de datos.
#### 2.1 Analisis de datos faltantes (nulos) y duplicados.
- La integridad de los datos previene sesgos algoritmicos, al tratarse de un modelo orientado a la salud, tenemos que tener la mejor calida de datos posibles.

In [40]:
# Valores nulos.
print(f"=" *80)
print(f"total de nulos: {df.isnull().sum()}")
# Valores duplicados.
print(f"=" *80)
duplicated = df.duplicated().sum()
print(f"Valores duplicados encontrados: {duplicated}")

if duplicated > 0:
    print(f"=" *80)
    print(f"Columna(s) donde se encuentra el valor duplicado") # Al tratarse del sector salud, siempre me interesara saber que dato es el que esta duplicado.
    display(df[df.duplicated()])

total de nulos: age         0
sex         0
cp          0
trestbps    0
chol        0
fbs         0
restecg     0
thalach     0
exang       0
oldpeak     0
slope       0
ca          0
thal        0
target      0
dtype: int64
Valores duplicados encontrados: 1
Columna(s) donde se encuentra el valor duplicado


,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
164,38,1,2,138,175,0,1,173,0,0.0,2,4,2,1


**Interpretación: Observamos que el data set no cuenta con valores nulos, y encontramos un valor duplicado en la columna 164, al tratarse de un data set pequeño, la mejor opcion sera eliminar el duplicado.

In [41]:
# Eliminar duplicado

df.drop_duplicates()
print(f"Valores duplicados despues de limpieza: {df.duplicated().sum()}")

Valores duplicados despues de limpieza: 1


### 3. Estadisticas descriptivas y target
#### 3.1 Analisis de la variable objetivo (target) 
Fundamentacion: En problemas de clasificacion un desbalance de clases podria invalidad la metrica accuracy